internal/pkg/v2/infrastructure/redis/reading.go

```diff
@@ -0,0 +1,72 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package redis
+
+import (
+	"encoding/json"
+
+	"github.com/edgexfoundry/edgex-go/internal/pkg/common"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
+
+	model "github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+
+	"github.com/gomodule/redigo/redis"
+
+	"github.com/google/uuid"
+)
+
+const ReadingsCollection = "v2:reading"
+
+var emptyBinaryValue = make([]byte, 0)
+
+// Add a reading to the database
+func addReading(conn redis.Conn, r model.Reading) (id string, err error) {
+	var m []byte
+	var baseReading *model.BaseReading
+	switch newReading := r.(type) {
+	case model.BinaryReading:
+		// Clear the binary data since we do not want to persist binary data to save on memory.
+		newReading.BinaryValue = emptyBinaryValue
+
+		baseReading = &newReading.BaseReading
+		checkReadingValue(baseReading)
+		m, err = json.Marshal(newReading)
+	case model.SimpleReading:
+		baseReading = &newReading.BaseReading
+		checkReadingValue(baseReading)
+		m, err = json.Marshal(newReading)
+	default:
+		return "", db.ErrUnsupportedReading
+	}
+
+	if err != nil {
+		return id, err
+	}
+	// use the SET command to save reading as blob
```

Comment:
V1 has following, which I assume was for adding a reading to existing Event, which we don't support, correct?
```
	if tx {
		_ = conn.Send("MULTI")
	}
....
	if tx {
		_, err = conn.Do("EXEC")
	}
```

Reply:
Yes, in V1 the `tx = true`case is invoked while calling the `AddReading` function(in `internal/pkg/db/redis/data.go`) used by the POST Reading API. As we don't have the POST Reading API in V2, we would not need the `tx=true` case, as the `addReading` function is always called in `addEvent`.